feat: add booking audit action services (lib/actions only)

hariombalhara opened the pull request:
## What does this PR do?

Extracts the `lib/actions` folder from the `booking-audit-more-infra`(#25125) branch as a standalone PR. This is a partial extraction to enable incremental review and merging of the booking audit infrastructure.

This PR adds audit action services that handle different booking lifecycle events. Each service follows a versioned schema pattern for data storage and migration support.

**New action services added:**
- AcceptedAuditActionService - tracks booking acceptance
- AttendeeAddedAuditActionService - tracks when attendees are added
- AttendeeNoShowUpdatedAuditActionService - tracks attendee no-show status changes
- AttendeeRemovedAuditActionService - tracks when attendees are removed
- CancelledAuditActionService - tracks booking cancellations
- HostNoShowUpdatedAuditActionService - tracks host no-show status changes
- LocationChangedAuditActionService - tracks location changes
- ReassignmentAuditActionService - tracks host reassignments
- RejectedAuditActionService - tracks booking rejections
- RescheduleRequestedAuditActionService - tracks reschedule requests
- RescheduledAuditActionService - tracks completed reschedules

**Supporting files:**
- `common/changeSchemas.ts` - Zod schemas for old→new value tracking

Requested by: @hariombalhara

## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [x] I have updated the developer docs in /docs if this PR makes changes that would require a [documentation change](https://cal.com/docs). N/A - infrastructure code only
- [ ] I confirm automated tests are in place that prove my fix is effective or that my feature works. (Tests will be added with the full infrastructure PR)

## How should this be tested?

This PR adds service classes that are not yet wired into the application. They will be integrated when the rest of the booking-audit infrastructure is merged.

For now, verification is limited to:
1. Type-check passes: `yarn type-check:ci --force`
2. Lint passes (verified via pre-commit hooks)

## Human Review Checklist

- [ ] Verify the interface changes in `IAuditActionService.ts` are acceptable
- [ ] Confirm `ReassignmentAuditActionService` import from `@calcom/features/users/repositories/UserRepository` resolves correctly
- [ ] Review the versioning strategy in each service (all currently V1 with migration scaffolding)
- [ ] Confirm this partial extraction approach is acceptable for incremental merging

## Checklist

- [x] I have read the [contributing guide](https://github.com/calcom/cal.com/blob/main/CONTRIBUTING.md)
- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have checked if my changes generate no new warnings

---
Link to Devin run: https://app.devin.ai/sessions/c332eef53fa34c43ac7637cba1f9e7d3

devin-ai-integration[bot] committed:
feat: add booking audit action services for lib/actions folder

Extract lib/actions folder changes from booking-audit-more-infra branch:
- AcceptedAuditActionService
- AttendeeAddedAuditActionService
- AttendeeNoShowUpdatedAuditActionService
- AttendeeRemovedAuditActionService
- CancelledAuditActionService
- CreatedAuditActionService (modified)
- HostNoShowUpdatedAuditActionService
- IAuditActionService (modified)
- LocationChangedAuditActionService
- ReassignmentAuditActionService
- RejectedAuditActionService
- RescheduleRequestedAuditActionService
- RescheduledAuditActionService

Also includes common/changeSchemas.ts required for type-checks to pass.

Co-Authored-By: hariom@cal.com <hariombalhara@gmail.com>

hariombalhara committed:
refactor: update AttendeeRemovedAuditActionService to correctly filter removed attendees

- Adjusted the logic in getDisplayJson to ensure only attendees present in the old list but not in the new list are included in removedAttendees.
- Removed unnecessary fields from the returned display data structure, streamlining the output.

Udit-takkar commented on packages/features/booking-audit/lib/actions/AttendeeRemovedAuditActionService.ts:
```diff
@@ -0,0 +1,82 @@
+// Module-level because it is passed to IAuditActionService type outside the class scope
+const fieldsSchemaV1 = z.object({
+    removedAttendees: StringArrayChangeSchema,
+});
```
what does `removedAttendees: StringArrayChangeSchema` in  AttendeeRemovedAuditActionService do? The schema here just tracks new attendee list.

hariombalhara committed:
refactor: update attendee tracking in booking audit services

- Renamed fields from `addedAttendees` and `removedAttendees` to `attendees` in both `AttendeeAddedAuditActionService` and `AttendeeRemovedAuditActionService`.
- Enhanced documentation to clarify how the `old` and `new` states of attendees are represented.
- Adjusted logic in `getDisplayJson` methods to correctly compute added and removed attendees based on the new structure.
